compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/ExpressionsConverter.kt

```diff
@@ -733,7 +733,7 @@ class ExpressionsConverter(
         }
         return if (whenRefWithSubject != null) {
             buildEqualityOperatorCall {
-                source = whenCondition.toFirSourceElement()
+                source = whenCondition.toFirSourceElement(FirFakeSourceElementKind.WhenCondition)
```

Comment:
What about `RawFirBuilder`?

Reply:
Ah, maybe it's done here? https://github.com/JetBrains/kotlin/blob/master/compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/PsiConversionUtils.kt#L36

Reply:
Yeah, `RawFirBuilder` seems to be handling this correctly already.